org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/services/AnnotationUtil.java

```diff
@@ -31,13 +31,15 @@ public static void findDelegateSegments(Class<?> clazz, Set<Class<?>> visited, C
 	}
 
 	public static boolean isDelegateMethod(Method method) {
-		JsonDelegate jsonDelegate = method.getAnnotation(JsonDelegate.class);
-		if (jsonDelegate != null) {
-			if (!(method.getParameterCount() == 0 && method.getReturnType().isInterface())) {
-				throw new IllegalStateException(
-						"The method " + method.toString() + " is not a proper @JsonDelegate method.");
+		if (!method.isSynthetic()) {
```

Comment:
That wouldn't change anything in terms of performance. Regarding readability, I'd say it's a matter of taste.